_pytest/capture.py

```diff
@@ -252,6 +252,8 @@ def writelines(self, linelist):
         self.write(data)
 
     def __getattr__(self, name):
+        if name == 'name':
+            return repr(self.buffer)
         return getattr(object.__getattribute__(self, "buffer"), name)
```

Comment:
Let's see if it is covered by some test(s): https://github.com/pytest-dev/pytest/pull/2614.